middleware/src/middleware.go

```diff
@@ -159,9 +173,82 @@ func (middleware *Middleware) rpcLoop() {
 	}
 }
 
+// updateCheckLoop repeatedly checks for information about new Base firmware updates
+// When an update is available it's
+func (middleware *Middleware) updateCheckLoop() {
+	// This time is chosen arbitrary, but the time should be not too high for users to be notified
+	// not to long after the update release and not too low to avoid to frequent update checks.
+	const timeBetweenUpdateChecks time.Duration = 30 * time.Minute
+
+	for {
+		updateInfo, err := getBaseUpdateInfo(middleware.environment.GetFirmwareUpdateInfoURL())
+		if err != nil {
+			log.Printf("Could not GET update info: %s\n", err)
+			time.Sleep(timeBetweenUpdateChecks)
+			continue
+		}
+
+		newVersion, err := semver.NewSemVerFromString(updateInfo.Version)
+		if err != nil {
+			log.Printf("Could not parse update info version as SemVer: %s\n", err)
+			time.Sleep(timeBetweenUpdateChecks)
+			continue
+		}
+
+		if !middleware.baseVersion.AtLeast(newVersion) {
+			log.Printf("A Base firmware update is available from version %s to %s.\n", middleware.baseVersion.String(), newVersion.String())
+			middleware.baseUpdateAvailable.UpdateAvailable = true
+			middleware.baseUpdateAvailable.UpdateInfo = updateInfo
+			middleware.events <- []byte(rpcmessages.OpBaseUpdateIsAvailable)
+		}
+
+		time.Sleep(timeBetweenUpdateChecks)
+	}
+}
+
+// getBaseUpdateInfo GETs a JSON file over HTTP which includes information about the update.
+func getBaseUpdateInfo(url string) (updateInfo rpcmessages.UpdateInfo, err error) {
+	client := http.Client{
+		Timeout: 15 * time.Second, // timeout is chosen arbitrary, but should account for (very) slow tor connections.
```

Comment:
The HTTP connection timeout of 15 seconds is chosen arbitrary. It should account for very slow tor connections. Let me know what you think.

Reply:
Going with 15 seconds here.